tests/js/client/shell/shell-dump-integration.js

```diff
@@ -593,6 +618,45 @@ function dumpIntegrationSuite () {
         } catch (err) {}
       }
     },
+
+    testDumpAllDatabasesWithOverwrite: function () {
+      let path = fs.getTempFile();
+      try {
+        let args = ['--all-databases', 'true'];
+        for (let i = 1; i < 3; ++i) {
+          db._createDatabase("databaseTest" + i);
+          db._useDatabase("databaseTest" + i);
+          db._create("collectionTest" + i);
+          db["collectionTest" + i].insert({});
+          db._useDatabase("_system");
+        }
+        runDump(path, args, 0);
+
+        // run the dump a second time, to overwrite all data in the target directory
+        args.push('--overwrite');
+        args.push('true');
+
+
+        let tree = runDump(path, args, 0);
+        for (let i = 1; i < 3; ++i) {
+          const dbName = "databaseTest" + i;
+          db._useDatabase(dbName);
+          assertNotEqual(-1, tree.indexOf(dbName));
+          checkDumpJsonFile(dbName, fs.join(path, dbName), db._id());
+          checkCollections(tree, path, dbName);
+          db._useDatabase("_system");
+        }
+      } finally {
+        try {
+          fs.removeDirectory(path);
+          db._useDatabase("_system");
+          for (let i = 1; i < 3; ++i) {
+            db._dropDatabase("databaseTest" + i);
+          }
+        } catch (err) {
+        }
```

Comment:
Having one global try...catch is not good here, because it means that there can still be leftovers.
For example, when the "removeDirectory" call fails and throws, the databases will not be cleaned up. The same is true if an error happens inside the cleanup loop.
```suggestion
        try {
          fs.removeDirectory(path);
        } catch (err) {}
        db._useDatabase("_system");
        for (let i = 1; i < 3; ++i) {
          try {
            db._dropDatabase("databaseTest" + i);
          } catch (err) {}
        }
```